frontend/src/lib/query/createTableColumnsFromQuery.ts

```diff
@@ -306,8 +301,6 @@ const fillRestAggregationData = (
 		if (!isEqual) {
 			equalQueriesByLabels.push(column.field);
 		}
-
-		column.data.push(parseFloat(targetSeria.values[0].value).toFixed(2));
```

Comment:
This code is executed only when there are multiple formulas. It checks if there are similar labels present in other formulas and, if found, adds them to the corresponding column data in the table.

However, since we're already looping through the series, which is responsible for adding data to its columns, adding it here may be redundant and may cause duplicated data to be added in the column for formulas (i.e. if the series contains 28 data point's then it will be 56).

The line 307 i.e. `equalQueriesByLabels.push(column.field);` is crucial. It ensures that no additional rows are added to the table for similar labels across all formulas here is how this check is applied: 
https://github.com/SigNoz/signoz/blob/e6e377beff550b5505a708cedd8688055f15b0cf/frontend/src/lib/query/createTableColumnsFromQuery.ts#L370

Reply:
@Rajat-Dabade we can add this as inline comment? PR might get lost in future during some refactor but inline comment would help

Reply:
Sure, I will add it.